### PR TITLE
Fix --var silently corrupting single-value lists (string gets char-split by Tera)

### PR DESCRIPTION
**Pull request by OmentaElvis**
# Bug

`parse_variables` decides whether a variable is a string or a list purely by counting how many times `--var key=...` was passed:

```rust
let value = if values.len() == 1 {
    json!(values[0])   // string
} else {
    json!(values)       // array
};
```

This means the number of CLI flags determines the type, not **user intent**. If a template expects a list but the user only has one item to pass, there is no way to tell parse_variables **"this is a list of one"**, it always collapses to a string.

Concretely, given this template:

```tera
{% for item in items | default(value=[]) %}
  - name: {{ item }}
  ...
{% endfor %}
```

```bash
laio start some-session --var items=widget
```

passes items to Tera as the string "widget". Tera's for loop iterates over characters of a string when it isn't given an array, so the loop silently runs once per letter `(w, i, d, g, e, t)`, generating multiple broken tmux windows/panes instead of the one intended for widget. No error is raised anywhere, the failure only shows up as a mangled session layout, which makes it non-obvious to diagnose.

The same problem exists in reverse too: any template relying on items being iterable will only work by accident, whenever the user happens to pass 2+ values.

# Fix

Add an explicit, opt-in array syntax: key[]=value. This lets the caller state their intent directly instead of leaving it to be inferred from flag count.

- `--var key=value` : unchanged. Still a plain string, exactly as before.
- `--var key=a --var key=b` : unchanged. Repeated keys still become an array.
- `--var key[]=value` : new. Forces array semantics even for a single occurrence, so ["value"] instead of "value".
- `--var key[]=a,b,c` : new. Also accepts a comma-separated list inside a single `[]` flag, so `["a", "b", "c"]`, avoids needing to repeat `--var key[]=...` for every item.
- `\,` inside a [] value escapes a literal comma (e.g. `key[]=foo\,bar,baz → ["foo,bar", "baz"]`).
Mixing `key=` and `key[]=` for the same key merges into a single array rather than erroring, since forcing users to pick one form for a given key added friction without much upside.

This makes the original bug report reproducible as an explicit, correct invocation:

```bash
laio start some-session --var tag=one --var tag=two --var 'items[]=widget'
```
or, more concisely:

```bash
laio start some-session --var tag=one --var tag=two --var 'items[]=widget,gadget,gizmo'
```

# Backward compatibility

No existing invocation changes meaning. key=value and repeated key=value behave identically to before; the new behavior only triggers when [] is present in the key. Covered explicitly by a test (`plain_value_with_comma_stays_literal`) asserting that commas in a non-bracketed value are still treated as a literal part of the string, not a delimiter.

## Defense in depth (optional, open to discussion)

Even with this fix, a template author can still write {% for x in y %} against a variable that turns out to be a plain string (e.g. if a user forgets []). Maybe add a small `as_array` Tera filter (wraps non-array values in a single-element array, arrays pass through unchanged) so templates can guard themselves independent of how the variable was parsed:

```
{% for item in items | default(value=[]) | as_array %}
```

# Testing

Added unit tests covering: plain single value, plain repeated key, bracket single value, bracket CSV split, bracket CSV across repeated flags, escaped comma, and literal comma in a non-bracketed value. All pass against the new implementation.

### CODE REVIEW

**Comment by OmentaElvis** [comment]
I updated the docs too to cover the new `--var` syntax.

File: src/common/config/variables.rs [hunk 1]
```diff
@@ -43,23 +75,92 @@ pub fn parse_variables(vars: &[String]) -> Result<HashMap<String, Value>> {
...
-            json!(values) // Multiple values become array
+            json!(acc.values[0])
         };
         map.insert(key, value);
     }
-
     Ok(map)
 }
+
+#[cfg(test)]
```
Comment (ck3mp3r) [inline src/common/config/variables.rs]: Please move these tests to the sibling file responsible for tests.

**Review by ck3mp3r — CHANGES_REQUESTED** [review]

**Review by ck3mp3r — COMMENTED** [review]
Let's discuss the suggestion for defence in depth using, I think it is worth it...

`{% for item in items | default(value=[]) | as_array %}`

**Comment by ck3mp3r** [comment]
# Review notes

The implementation is correct and backward-compatible. The items below are gaps in the docs and the tests.

## Blocker

- Move the new tests from `variables.rs` to the sibling file `variables_test.rs`. The repo uses sibling test files. An inline `mod tests` breaks the pattern.

## Docs gaps

- The error message at `variables.rs:31` says only `Expected format: key=value`. Update the message to include the `key[]=a,b,c` form.
- The docs do not state that `split_csv` trims each item. Add a sentence to the yaml reference.
- The docs do not state the behavior of `key[]=` with an empty value. The code produces `[""]`, not `[]`. Document this or change the behavior.
- The docs describe only `\,` as an escape. A literal backslash before a non-comma character is undocumented. Add a sentence or an example.

## Untested edge cases

- `key[]=a,,b` produces `["a", "", "b"]`. Add a test.
- `key=alpha,beta` (plain, comma literal) mixed with `key[]=gamma` produces `["alpha,beta", "gamma"]`. Add a test. The existing mix test uses a comma-free plain value.
- `key[][]=value` and `key[]foo=value` are not tested. Add a test for each.
- `  []  =value` (empty key after strip) is not tested. Add a test.

**Review by ck3mp3r — APPROVED** [review]

**Comment by OmentaElvis** [comment]
> The docs do not state the behavior of key[]= with an empty value. The code produces [""], not []. Document this or change the behavior.

I have been thinking that we treat `key[]=` as a special case that produces `[]` rather than `[""]`. To me it makes sense that the user intent is "_**I have this variable that is supposed to have values but right now it is empty**_". That closely meets user expectations because `[]` is not equal to `[""]` in any way and it will probably be very complicated for the user to verify that the variable value is not just an empty string.

They could always use Tera inbuilt `default(value=[])` but 99% of the time the user is likely to pass `key[]=` and expect it to produce an empty array. 

> Let's discuss the suggestion for defense in depth using, I think it is worth it...
 {% for item in items | default(value=[]) | as_array %}

I have added a simple tera filter `as_array`. It is opt-in guard that the users can use to safely guarantee that a variable is an array. For example if you have `{% for item in items | default(value=[]) | as_array %}` and you pass `--var items=foo`, as_array will guarantee that items is passed down as `["foo"]`. The original behavior was that items would receive "foo" and tera will just blindly iterate against every character of the word "foo". This is just good addition to harden existing templates from simple mistakes.

**Review by ck3mp3r — APPROVED** [review]
